pynamodb/attributes.py

```diff
@@ -72,6 +74,10 @@ def __init__(
     ) -> None:
```

Comment:
Do you think we could type `default`/`default_for_new` with the immutable types so that anyone using mypy gets earlier feedback?

Reply:
I tried but it gets messy.

At the least you'd need two overloads:
```python
ImmutableT = TypeVar('ImmutableT', str, int, ...)

class Attribute(Generic[T]):
  @overload
  def __init__(self: Attribute[ImmutableT], ..., default: ImmutableT | Callable[..., ImmutableT] | None, ...) -> None: ...
  
  @overload
  def __init__(self, ..., default: Callable[..., T] | None, ...) -> None: ...
```

But that [doesn't seem to work](https://github.com/python/mypy/issues/13993). In fact, even a simpler one where I explicitly single out `str` doesn't seem to work:
```python
class Attribute(Generic[T]):
  @overload
  def __init__(self: Attribute[str], ..., default: str | Callable[..., str] | None, ...) -> None: ...
  
  @overload
  def __init__(self, ..., default: Callable[..., T] | None, ...) -> None: ...
```

See [playground](https://mypy-play.net/?mypy=latest&python=3.10&gist=30f7918d00fbe9c4a86bfacea2d20da3): note how this line passes:
```python
list_attr3 = ListAttribute(default='foobar')
```

Reply:
gotcha - it's a shame but I suppose this is fine for now and we can hope for type improvements later